src/channel_internal.c

```diff
@@ -279,148 +353,166 @@ static int dequeue_operation(THANDLE(CHANNEL_INTERNAL) channel_internal, THANDLE
     PDLIST_ENTRY op_entry = DList_RemoveHeadList(&channel_internal_ptr->op_list);
 
     CHANNEL_OP* channel_op = CONTAINING_RECORD(op_entry, CHANNEL_OP, anchor);
-    if (pull_callback != NULL)
+    if (on_data_available_cb != NULL)
     {
-        /*Codes_SRS_CHANNEL_INTERNAL_43_112: [ channel_internal_pull shall store the correlation_id, pull_callback and pull_context in the obtained operation. ]*/
+        /*Codes_SRS_CHANNEL_INTERNAL_43_112: [ channel_internal_pull shall store the correlation_id, on_data_available_cb and pull_context in the obtained operation. ]*/
         THANDLE_INITIALIZE(RC_STRING)(&channel_op->pull_correlation_id, pull_correlation_id);
-        channel_op->pull_callback = pull_callback;
+        channel_op->on_data_available_cb = on_data_available_cb;
         channel_op->pull_context = pull_context;
     }
-    else if (push_callback != NULL)
+    else if (on_data_consumed_cb != NULL)
```

Comment:
>else [](http://example.com/codeflow?start=4&length=4)

Uh oh, should this `else` be here.  ie - it looks like we want to ignore `on_data_consumed_cb` if `on_data_available_cb` is set?  Is this correct?  I _think_ this is wrong, but I can't really review the rest of this function until I get my head wrapped around this

Reply:
Also, if on_data_available_cb is set, does push_context leak?

Reply:
The helper function `dequeue_operation` is used in push and pull. This check is mutually exclusive, so the `else` can be there even if it is not strictly necessary.